File: MyTest2/app/src/main/java/com/example/test/MainActivity.java
package com.example.test;

import androidx.appcompat.app.AppCompatActivity;

import android.opengl.Visibility;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    TextView labelImage1;
    ImageView imageIconBlack;
    ImageView imageIconBlue;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        labelImage1 = findViewById(R.id.label_image1);
        imageIconBlack = findViewById(R.id.image_icon_black);
        imageIconBlue = findViewById(R.id.image_icon_blue);
        labelImage1.setOnClickListener(click -> onClickForColorChange());
        imageIconBlack.setOnClickListener(view -> onClickForColorChange());
        imageIconBlue.setOnClickListener(view -> onClickForColorChange());
    }

    protected void onClickForColorChange() {
        String currentColor = getCurrentColor();
        changeColor(currentColor);
    }

    protected String getCurrentColor() {
        if (imageIconBlue.getVisibility() == View.GONE) return "black";
        else return "blue";
    }

    protected void changeColor(String currentColor) {
        if (currentColor.equals("black")) imageIconBlue.setVisibility(View.VISIBLE);
        else imageIconBlue.setVisibility(View.GONE);
    }
}

File: Tutorial1/app/src/main/java/com/example/tutorial1/MainActivity.kt
package com.example.tutorial1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.EditText

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    public void addClick(View v) {

        EditText number1 = (EditText) findViewByld(R.id.number1) ;
        EditText number2 = (EditText) findViewByld(R.id.number2) ;
        TextView result = (TextView) findViewByld(R.id.result) ;
        int n1 = Integer.parseInt(number1.getText().toString()) ;
        int n2 = Integer.parseInt(number2.getText().toString());
        result.setText(Integer.toString(n1 + n2));
    }
    public void subtractClick(View v) {
        EditText number1 =(EditText) findViewByld (R.id.number1);
        EditText number2 =(EditText) findViewByld (R.id.number2);
        TextView result =(TextView) findViewByld (R.id.result);
        int n1 = Integer . parseInt (number1.getText().toString());
        int n2 = Integer . parseInt (number2.getText().toString());
        result.setText(Integer.toString(n1 - n2));
    }
    public void multiplyClick(View v) {
        EditText number1 =(EditText) findViewByld (R.id.number1);
        EditText number2 =(EditText) findViewByld (R.id.number2);
        TextView result =(TextView) findViewByld (R.id.result);
        int n1 = Integer . parseInt (number1.getText().toString());
        int n2 = Integer . parseInt (number2.getText().toString());
        result.setText(Integer.toString(n1 * n2));
    }
    public void divideClick(View v) {
        EditText number1 =(EditText) findViewByld (R.id.number1);
        EditText number2 =(EditText) findViewByld (R.id.number2);
        TextView result =(TextView) findViewByld (R.id.result);
        int n1 = Integer . parseInt (number1.getText().toString());
        int n2 = Integer . parseInt (number2.getText().toString());
        result.setText(Integer.toString(n1 / n2));
    }
}